3 packages from cyrus-/relit at 0.2.0

charlesetc opened the pull request:
This pull-request concerns:
-`ppx_relit.0.2.0`: An implementation of Typed Literal Macros for Reason
-`relit_helper.0.2.0`: A helper library for those wishing to write TLMs using Relit



---
* Homepage: https://github.com/cyrus-/relit
* Source repo: git://github.com/cyrus-/relit.git
* Bug tracker: https://github.com/cyrus-/relit/issues

---
:camel: Pull-request generated by opam-publish v2.0.0

camelus commented:
##### :sunny: All lint checks passed <small>13805ecbc18395510eeba9d83cae8c873c73e29f</small>

* These packages passed lint tests: `ppx_relit.0.2.0`, `relit_helper.0.2.0`


---


##### :sun_behind_small_cloud: Installability check (9965 &rarr; 9965)

* new **broken** packages (2): `ppx_relit.0.2.0` `relit_helper.0.2.0`

mseri commented on packages/relit_helper/relit_helper.0.2.0/opam:
```diff
@@ -0,0 +1,30 @@
+  "dune"
+  "extlib"
+  "ppxlib" {= "0.3.1"}
+  "ocaml-migrate-parsetree" {= "1.0.11"}
```
Do you really need so strict version bounds?

kit-ty-kate commented:
As @mseri noticed I wonder why there are so many strict version bounds. It would be better to avoid those. Also the `dune` dependencies don't have the `{build}` attribute. Please add it.
And the `ocamlfind` dependency is legit for `ppx_relit` but not for `relit-reason` as far as I could see. Also the dependency is only a dependency on the executable so the `{build}` attribute can also be added.

charlesetc commented:
Thanks for the help! I've relaxed the restrictions a bit: we specifically do need ppxlib greater than or equal to 0.3.1, but probably not pinned so I've fixed that. Also added `{build}` to the dune dependencies. OCamlfind is a runtime dependency of relit so I think that should stay without the `{build}`? Anyways hope the tests pass and thanks again!

charlesetc commented:
Also removed the pin of ocaml-migrate-parsetree!

charlesetc commented:
How does it look now?

kit-ty-kate commented:
Good. Thanks!

charlesetc commented:
Thanks very much! :balloon:
